openstack/neutron/delete-basic.sls

```diff
@@ -18,65 +18,51 @@ user_domain_env delete:
+{% for each in ['flat','flat1','ext-net'] %}
+  {% set subnet_id = salt['cmd.run'](cmd='neutron ' + neutron_auth + ' subnet-list --name=' + each + ' --column id --format value', env={'OS_PROJECT_DOMAIN_ID': 'default', 'OS_USER_DOMAIN_ID': 'default'} if virl.mitaka else {} ) %}
+  {% if subnet_id %}
+  
+delete floating ips {{ each }}:
   cmd.run:
-    - name: neutron --os-tenant-name admin --os-username admin --os-password {{ ospassword }} --os-auth-url=http://{{ controllerip }}:5000/{{ kav }} port-list --format csv --column id | sed 1d | xargs -rn1 neutron --os-tenant-name admin --os-username admin --os-password {{ ospassword }} --os-auth-url=http://{{ controllerip }}:5000/{{ kav }} port-delete $1
-    - onfail:
-      - cmd: delete ports
-{% if virl.mitaka %}
+    - name: neutron {{ neutron_auth }} port-list --fixed_ips subnet_id={{ subnet_id }} --device_owner=network:floatingip --column device_id --format value | xargs -rn1 neutron {{ neutron_auth }} floatingip-delete
+    {% if virl.mitaka %}
     - require:
-      - environ: project_domain_env create
-      - environ: user_domain_env create
-{% endif %}
+      - environ: project_domain_env delete
+      - environ: user_domain_env delete
+    {% endif %}
 
-# delete floating ips
-delete floating ips:
+clear router-gateway {{ each }}:
   cmd.run:
-    - name: neutron --os-tenant-name admin --os-username admin --os-password {{ ospassword }} --os-auth-url=http://{{ controllerip }}:5000/{{ kav }} floatingip-list --format csv --column id | sed 1d | xargs -rn1 neutron --os-tenant-name admin --os-username admin --os-password {{ ospassword }} --os-auth-url=http://{{ controllerip }}:5000/{{ kav }} floatingip-delete $1
-{% if virl.mitaka %}
+    - name: neutron {{ neutron_auth }} port-list --fixed_ips subnet_id={{ subnet_id }} --device_owner=network:router_gateway --column device_id --format value | xargs -rn1 neutron {{ neutron_auth }} router-gateway-clear
+    {% if virl.mitaka %}
     - require:
-      - environ: project_domain_env create
-      - environ: user_domain_env create
-{% endif %}
+      - environ: project_domain_env delete
+      - environ: user_domain_env delete
+    {% endif %}
 
-clear ext-net router-gateway:
+delete ports {{ each }}:
   cmd.run:
-    - name: neutron --os-tenant-name admin --os-username admin --os-password {{ ospassword }} --os-auth-url=http://{{ controllerip }}:5000/{{ kav }} router-list -c id -f csv | grep -o '[a-fA-F0-9-]\{36\}' | xargs -rn1 neutron --os-tenant-name admin --os-username admin --os-password {{ ospassword }} --os-auth-url=http://{{ controllerip }}:5000/{{ kav }} router-gateway-clear
-    - onlyif: neutron --os-tenant-name admin --os-username admin --os-password {{ ospassword }} --os-auth-url=http://{{ controllerip }}:5000/{{ kav }} subnet-show ext-net
-{% if virl.mitaka %}
+    - name: neutron {{ neutron_auth }} port-list --fixed_ips subnet_id={{ subnet_id }} --column id --format value | xargs -rn1 neutron {{ neutron_auth }} port-delete
```

Comment:
Ports which do have device_id set, or which have device_owner starting with network (e.g. router gw, dhcp, router interface) should normally fail to get deleted by a normal port delete.

For ext-net, which is also the only network which may host router gateways, the proper thing to do is to clear all router gateways present in the system, and then reset them in create-basic.

Do the equivalent create-basic statements still re-set e.g. the gateways?